src/services/inlayHints.ts

```diff
@@ -463,28 +471,33 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
         return typeToInlayHintParts(signatureParamType);
     }
 
-    function printTypeInSingleLine(type: Type) {
+    function typeToInlayHintParts(type: Type): InlayHintDisplayPart[] | string {
         const flags = NodeBuilderFlags.IgnoreErrors | NodeBuilderFlags.AllowUniqueESSymbolType | NodeBuilderFlags.UseAliasDefinedOutsideCurrentScope;
-        const printer = createPrinterWithRemoveComments();
 
-        return usingSingleLineStringWriter(writer => {

```

Comment:
Why is this bit being removed? Don't we need to ensure that these are single lines?

Reply:
IIRC, I concluded that this didn't have anything logic inside it that would not already be covered by `typeToString`. When types are written in error diagnostics etc they are single line. I could miss something so if you think this is wrong - I could revisit this.

Reply:
I think that even if the logic is already being done in `typeToString` it's better to be explicit here and keep the current code (also because we don't want to rely on `typeToString`s behavior).

Reply:
I reverted this change 👍